From fd900f79b9219790478863acff9c53459bbbf7b0 Mon Sep 17 00:00:00 2001
From: Lorenzo Bianconi <lorenzo@kernel.org>
Date: Sun, 6 Mar 2022 20:34:11 +0100
Subject: [PATCH 5/9] nl80211: Radar background flag setting

Allow background radar detection flag to be set when specifying a
channel. This is a preliminary change to introduce radar/CAC background
detection support.

Signed-off-by: Lorenzo Bianconi <lorenzo@kernel.org>
Signed-off-by: Manish Dharanenthiran <quic_mdharane@quicinc.com>
---
 src/drivers/driver.h         | 5 +++++
 src/drivers/driver_nl80211.c | 5 +++++
 2 files changed, 10 insertions(+)

diff --git a/src/drivers/driver.h b/src/drivers/driver.h
index d8cbae7..0a3a10e 100644
--- a/src/drivers/driver.h
+++ b/src/drivers/driver.h
@@ -810,6 +810,11 @@ struct hostapd_freq_params {
 	 */
 	struct ieee80211_edmg_config edmg;
 
+	/**
+	 * radar_background - Whether radar/CAC background is requested
+	 */
+	bool radar_background;
+
 	/**
 	 * he_6ghz_reg_pwr_type - 6G regulatory power mode
 	 * Since many operation related to channel for 6G depends on the
diff --git a/src/drivers/driver_nl80211.c b/src/drivers/driver_nl80211.c
index d7b9508..7341080 100644
--- a/src/drivers/driver_nl80211.c
+++ b/src/drivers/driver_nl80211.c
@@ -5010,6 +5010,8 @@ static int nl80211_put_freq_params(struct nl_msg *msg,
 	wpa_printf(MSG_DEBUG, "  * he_enabled=%d", freq->he_enabled);
 	wpa_printf(MSG_DEBUG, "  * vht_enabled=%d", freq->vht_enabled);
 	wpa_printf(MSG_DEBUG, "  * ht_enabled=%d", freq->ht_enabled);
+	wpa_printf(MSG_DEBUG, "  * radar_background=%d",
+		   freq->radar_background);
 
 	hw_mode = ieee80211_freq_to_chan(freq->freq, &channel);
 	is_24ghz = hw_mode == HOSTAPD_MODE_IEEE80211G ||
@@ -5099,6 +5101,9 @@ static int nl80211_put_freq_params(struct nl_msg *msg,
 				NL80211_CHAN_NO_HT))
 			return -ENOBUFS;
 	}
+	if (freq->radar_background)
+		nla_put_flag(msg, NL80211_ATTR_RADAR_BACKGROUND);
+
 	return 0;
 }
 
-- 
2.17.1

